bin/export_etas_overlap_report.rb

```diff
@@ -13,41 +13,83 @@
 
 Services.mongo!
 
-def open_report(org, date, path)
-  File.open("#{path}/#{org}_#{date}.tsv", "w")
-end
+# Generates overlap reports for 1 or all members
+class EtasMemberOverlapReport
+  attr_accessor :reports, :date_of_report, :report_path, :organization
 
-if __FILE__ == $PROGRAM_NAME
-  date_of_report = Time.now.strftime("%Y-%m-%d")
-  report_path = Settings.etas_overlap_reports_path
-  Dir.mkdir(report_path) unless File.exist?(report_path)
-  reports = {}
+  def initialize(organization = nil)
+    @reports = {}
+    @date_of_report = Time.now.strftime("%Y-%m-%d")
+    @report_path = Settings.etas_overlap_reports_path || "tmp_reports"
+    Dir.mkdir(@report_path) unless File.exist?(@report_path)
+    @organization = organization
+  end
 
-  BATCH_SIZE = 10_000
-  waypoint = Utils::Waypoint.new(BATCH_SIZE)
-  logger = Logger.new($stderr)
-  logger.info "Starting #{Pathname.new(__FILE__).basename}. Batches of #{ppnum BATCH_SIZE}"
+  def open_report(org, date)
+    File.open("#{report_path}/#{org}_#{date}.tsv", "w")
+  end
 
-  org = ARGV.shift
-  ClusterOverlap.matching_clusters(org).each do |c|
-    ClusterOverlap.new(c, org).each do |overlap|
-      waypoint.incr
+  def report_for_org(org)
+    unless reports.key?(org)
+      reports[org] = open_report(org, date_of_report)
+    end
+    reports[org]
+  end
+
+  def clusters_with_holdings
+    if organization.nil?
+      Cluster.where("holdings.0": { "$exists": 1 }).no_timeout
+    else
+      Cluster.where("holdings.organization": organization).no_timeout
+    end
+  end
+
+  def write_record(holding, format, access, rights)
+    etas_record = ETASOverlap.new(ocn: holding[:ocn],
+                    local_id: holding[:local_id],
+                    item_type: format,
+                    access: access,
+                    rights: rights)
+    report_for_org(holding[:organization]).puts etas_record
+  end
+
+  def missed_holdings(cluster, holdings_matched)
+    if organization.nil?
+      cluster.holdings - holdings_matched.to_a
+    else
+      cluster.holdings.group_by(&:organization)[organization] - holdings_matched.to_a
+    end
+  end
+
+  def write_overlaps(cluster, organization)
```

Comment:
I'd like to see the different outputs of write_overlap (what is passed to write_record and what is returned) explained in a comment.